python/triton/language/math.py

```diff
@@ -1,17 +1,18 @@
 import functools
 import os
 
+from ..common.build import is_hip
 from . import core
 
 
 @functools.lru_cache()
 def libdevice_path():
-    import torch
     third_party_dir = os.path.join(os.path.dirname(os.path.abspath(__file__)), "..", "third_party")
-    if torch.version.hip is None:
-        default = os.path.join(third_party_dir, "cuda", "lib", "libdevice.10.bc")
+    if is_hip():
+        default = os.path.join(third_party_dir, "hip", "lib", "bitcode", "cuda2gcn.bc")
     else:
-        default = ''
+        default = os.path.join(third_party_dir, "cuda", "lib", "libdevice.10.bc")
+
```

Comment:
Can this be a device backend query for custom device bitcode be added instead of a binary between NVIDIA and AMD?

Reply:
this is a similar problem to the gpu dialect query issue [here](https://github.com/openai/triton/pull/1983#pullrequestreview-1563786382). Let me see what I can do.

Reply:
Will be address in followup pr.